### ISSUE TITLE
`@include all-X-sprites` references non-relative url for sprite image

### ISSUE THREAD

**Issue by dlo** [Compass/compass#834]
My directory structure looks like this:

```
├── scss
├── css
└── img
    └── social
```

The sprites are in my social directory. Here is the relevant SCSS:

```
@import "social/*.png";
@include all-social-sprites;
```

This compiles to something like:

```
.social-sprite, .social-facebook, .social-flag, .social-linkedin, .social-mail, .social-twitter {
  background: url('/media/img/social-sf479c8398f.png') no-repeat;
}

...
```

But when I deploy to production, my media lives on S3, on a different domain and path, so this URL breaks. A working path would be relative, like the following:

```
.social-sprite, .social-facebook, .social-flag, .social-linkedin, .social-mail, .social-twitter {
  background: url('../img/social-sf479c8398f.png') no-repeat;
}
```

Thanks!

**Comment by scottdavis** [comment]
whats in your compass config file?

**Comment by dlo** [comment]
Here it is!

```
http_path = "/"
css_dir = "media/css"
sass_dir = "media/scss"
images_dir = "media/images"
fonts_dir = "media/fonts"
javascripts_dir = "media/js"
line_comments = false
relative_assets = true
```

**Comment by scottdavis** [comment]
i think i found the problem is this simple enough of a project that i can use it to debug this? or can you make me one?

**Comment by scottdavis** [comment]
i think the issue has todo with this function

https://github.com/chriseppstein/compass/blob/stable/lib/compass/sass_extensions/functions/urls.rb#L148

**Comment by scottdavis** [comment]
actually try setting your `http_generated_images_path` in the config file first

**Comment by dlo** [comment]
Just set `http_generated_images_path`, didn't seem to have any affect. The generated URL seems to be generated regardless of its value.

**Comment by dlo** [comment]
Still need a sample project?

**Comment by scottdavis** [comment]
please

**Comment by chriseppstein** [comment]
@dlo @scottdavis status?

**Comment by HNygard** [comment]
Have a look a my config.rb in #951. Setting `generated_images_dir` and `http_generated_images_path`

**Comment by chriseppstein** [comment]
You can turn on `relative_assets = true` in your config.

I found a copy/paste error in the function. That plus a few other fixes in this function over the previous few months make me think this is fixed in 1.0. file a new bug with details if you're still having issues.
